File: 2106-find-greatest-common-divisor-of-array/find-greatest-common-divisor-of-array.cpp
class Solution {
public:

    int myFunction(int a,int b){

        while(a != 0 && b != 0){

            if(a >= b){
                a = a -b;
            }
            else if(b > a){
                b = b - a;
            }
        }

        return a == 0 ? b : a;
    }
    int findGCD(vector<int>& nums) {
        int maxi = INT_MIN;
        int mini = INT_MAX;
       for(int i=0;i<nums.size();i++){
        if(nums[i] > maxi){
            maxi = nums[i];
        }
        if(nums[i] < mini){
            mini = nums[i];
        }
       }


       int ans  = myFunction(mini,maxi);
        
    return ans;
    }
};

File: 2106-find-greatest-common-divisor-of-array/find-greatest-common-divisor-of-array.py
class Solution:

    def myFunction(self,a, b):
        while a != 0 and b != 0 :
            if a >= b:
                a = a -b
            elif b > a:
                b = b- a
        
        if a == 0:
            return b
        if b == 0:
            return a

    def findGCD(self, nums: List[int]) -> int:
        mini = float('inf')
        maxi = float('-inf')
        for x in range(0,len(nums)):
            if nums[x] <= mini :
                mini = nums[x]
            if nums[x] >= maxi:
                maxi = nums[x]

        return  self.myFunction(mini,maxi)
        